backend-rust/src/graphql_api.rs

```diff
@@ -4917,11 +4916,23 @@ pub struct CredentialsUpdated {
 }
 
 #[derive(SimpleObject, serde::Serialize, serde::Deserialize)]
+#[graphql(complex)]
 pub struct DataRegistered {
-    decoded:     DecodedText,
     data_as_hex: String,
 }
 
+#[ComplexObject]
+impl DataRegistered {
+    async fn decoded(&self) -> ApiResult<DecodedText> {
+        let decoded_data = hex::decode(&self.data_as_hex).map_err(|e| {
+            error!("Invalid hex encoding {:?} in a controlled environment", e);
```

Comment:
It is fine as it is.

Long-term we plan to have a middleware layer to trace all the internal errors and report them consistently so (not implemented yet).
https://github.com/Concordium/concordium-scan/issues/270